tests/testsuite/update.rs

```diff
@@ -1823,14 +1812,7 @@ fn update_precise_breaking_pre_release_cannot_upgrade_nonexplicit_version_req()
         .masquerade_as_nightly_cargo(&["update-precise-breaking"])
         .with_status(101)
         .with_stderr_data(str![[r#"
-[UPDATING] `dummy-registry` index
-[ERROR] failed to select a version for the requirement `pre = "^0.1"`
-candidate versions found which didn't match: 0.2.0-beta
-location searched: `dummy-registry` index (which is replacing registry `crates-io`)
-required by package `foo v0.0.1 ([ROOT]/foo)`
-if you are looking for the prerelease package it needs to be specified explicitly
-    pre = { version = "0.2.0-beta" }
-perhaps a crate was updated and forgotten to be re-vendored?
+[ERROR] New requirement ^0.2 is invalid, because it doesn't match 0.2.0-beta
```

Comment:
@epage I'm curious what you think of this.

Reply:
Lower casing fixed. Awaiting feedback on the error case.

Reply:
Why is this an error?
- manifest points to `0.1`
- `0.2.0-beta` is published
- user does `cargo update --precise 0.2.0-beta`

We gave it a valid version that it a breaking change from the current one, so it seems like we should just use that as our version requirement.

The fact that the error doesn't make sense from the users perspective ("I never said the version req was `^0.2` so why is it trying that?") I think helps highlight that.

Reply:
Looks like its only an error because the user's req was `0.1` instead of `0.1.0`?  In that case, I think its reasonable for us to act as if it was `0.1.0`.